### ISSUE TITLE
[feature] Use dictionaries to express arch values in settings.yml

### ISSUE THREAD

**Issue by liarokapisv** [conan-io/conan#13313]
### What is your suggestion?

Hi there, we are an embedded-focused company, and we heavily use cross-building for our products.

Due to microcontrollers having different fpu & mcu architectures, expressing `arch` using the default settings is unsatisfactory for us. 

Instead, we maintain toolchain-specific `arch` dictionary-values that express each toolchain's arch-flags. Then we use these in our toolchain profiles.

For example, we maintain an arm-none-eabi-gcc recipe. arm-gcc has native `-mcpu`, `-mfpu`, `-mfloat-abi` flags along with `-mthumb` to change the instruction set. We maintain a `gcc-arm` "arch" that basically looks like this:

```
arch:
  //...
  gcc-arm:
    instructions:
      - arm
      - thumb
    cpu:
      - cortex-m0plus
      - cortex-m4
      - cortex-m7
    fpu:
      - auto
      - fpv4-sp-d16
      - fpv5-d16
      - fpv5-sp-d16
```

Ideally, we would like to simply add this to the new `settings_user.yml` file and be done with it.
Unfortunately, this does not really work due to `arch` being a list.

We found out that expressing `arch` as a dictionary in `settings.yml` is backwards-compatible with the list approach and also allows us
to maintain such structured `arch` values separately in `settings_user.yml`. Ι think it would be a useful change to others as well.

TLDR:

I am suggesting changing:

```
arch: ["x86", " x86_64", "ppc32be", "ppc32", ...]
```
to

```
arch: 
  x86:
  x86_64:
  ppc32be:
  ppc32:
  ...
```

### Have you read the CONTRIBUTING guide?

- [X] I've read the CONTRIBUTING guide

**Comment by liarokapisv** [comment]
Maybe a better approach to this, is actually tackling the reason that we can't merge the structured arch values with the ones in settings.yml. The underlying reason seems to be that it's inherently a list vs a dictionary, but actually converting to a dictionary seems to work just fine from our testing so far. Maybe a conversion to a dictionary could be done behind the scenes?

**Comment by memsharded** [comment]
https://github.com/conan-io/conan/pull/13323 merged, this will be in 2.0.2
